core/src/test/scripts/unit_tests_validate_data.py

```diff
@@ -1298,26 +1283,27 @@ def test_unnecessary_and_wrong_stable_id(self):
 
 
 class HeaderlessClinicalDataValidationTest(PostClinicalDataFileTestCase):
-    
+
     """Superclass for validating headerless clinical data."""
-    
+
     def test_headerless_clinical_sample(self):
         self.logger.setLevel(logging.WARNING)
         record_list = self.validate('data_clinical_sam_no_hdr.txt', 
                                     validateData.SampleClinicalValidator, None, True)
-        # we expect 3 errors or warnings for 2 new sample-level attributes in data
-        # and 1 for not being able to parse the header in the clinical file
-        self.assertEqual(len(record_list), 3)
-    
+        # we expect 1 error for not being able to parse the header
+        self.assertEqual(len(record_list), 1)
+
     def test_headerless_clinical_patient(self):
         self.logger.setLevel(logging.WARNING)
         record_list = self.validate('data_clinical_pat_no_hdr.txt', 
                                     validateData.PatientClinicalValidator, None, True)
-        # we expect 8 errors or warnings: 2 for new patient-level attributes,
-        # 1 for the sample attribute CANCER_TYPE in a patient-level file, 2
-        # for missing survival status attributes, 4 for patients with no samples
-        # and 1 for not being able to parse the header in the clinical file
-        self.assertEqual(len(record_list), 10)
+        # we expect 9 errors or warnings:
+        # 1 for not being able to parse the header in the clinical file,
+        # 1 for the sample attribute CANCER_TYPE in a patient-level file,
+        # 1 for CANCER_TYPE being in DEFINED_SAMPLE_ATTRIBUTES
+        # 2 for missing survival status attributes and
+        # 4 for patients with no samples
+        self.assertEqual(len(record_list), 9)
```

Comment:
I agree. But the design of this test case is not the issue I intended to fix in this pull request.